test/secmemtest.c

```diff
@@ -75,12 +75,31 @@ static int test_sec_mem(void)
     TEST_ptr_null(OPENSSL_secure_malloc((size_t)-1));
     TEST_true(CRYPTO_secure_malloc_done());
 
+    /*-
+     * There was also a possible infinite loop when the number of
+     * elements was 1<<31, as |int i| was set to that, which is a
+     * negative number. However, it requires minimum input values:
+     *
+     * CRYPTO_secure_malloc_init((size_t)1<<34, (size_t)1<<4);
+     *
+     * Which really only works on 64-bit systems, and even then the
+     * code attempts to allocate 16 GB secure memory arena. Linux
+     * can deal with this better than other Unixy OS's (e.g. MacOS)
```

Comment:
> surpassing the [memorylocked] limit is privileged operation

Or more specifically increasing the limit past certain boundary is. I mean it's not like if you are root, then limit is simply ignored, no, it's still enforced. It's just that root has an option to increase the limit to arbitrarily high value. This by the way calls for following question. Since failure to mlock is not considered fatal, is application entitled to expect best effort from sh_init() which would entail attempt to raise RLIMIT_MEMLOCK [if deemed necessary]? Or should setting the limit be left to application?

For reference, default memorylocked limit on Linux is 64KB, even for root. What is typical secmem pool would you expect application asks for?